zerver/lib/actions.py

```diff
@@ -2524,14 +2524,15 @@ def gather_subscriptions_helper(user_profile):
 
     stream_ids = [sub["recipient__type_id"] for sub in sub_dicts]
 
-    stream_dicts = get_active_streams(user_profile.realm).select_related(
+    all_active_streams = get_active_streams(user_profile.realm)
+    stream_dicts = all_active_streams.select_related(
         "realm").filter(id__in=stream_ids).values(
         "id", "name", "invite_only", "realm_id", "realm__domain", "email_token", "description")
     stream_hash = {}
     for stream in stream_dicts:
         stream_hash[stream["id"]] = stream
 
-    all_streams = list(get_all_realm_streams(user_profile.realm).values())
+    all_streams = list(all_active_streams.values())
```

Comment:
This is still making an extra query. If you remove the `filter(id_in=stream_ids)` from line 2529 then `stream_dicts` will have all active streams and you can use that for `never_subscribed` stream IDs